### PR TITLE
Add fallback rendering for hallucinated inline chart shorthand

### PR DESCRIPTION
**Pull request by NKoech123**
### Summary
Adds a best-effort fallback renderer that detects and renders an inline chart when an agent emits a hallucinated `/word ... labels=[...] data=[...]` line in chat instead of the documented ` ```embed ` fence.

### Problem
Models occasionally regress to typing chart tool parameters (`type`/`title`/`labels`/`data`) as plain chat text instead of calling the chart action or emitting a proper `/chart` embed fence. Previously, that line rendered as inert literal text with no chart at all, giving users no useful output.

### Solution
Introduce a generic detector and parser for this shorthand pattern that is not tied to any specific tool name, plus a lightweight React renderer that produces an inline chart from the parsed data. Malformed input (mismatched lengths, negative values, oversized/deeply nested arrays) safely falls back to plain text instead of throwing, and content inside fenced or indented code blocks is correctly skipped.

### Key Changes
- Added `legacy-chart-shorthand.tsx` with `looksLikeLegacyChartShorthand`, `parseLegacyChartShorthand`, `wrapLegacyChartShorthandLines`, and the `LegacyChartShorthandChart` component to detect, parse, and render the shorthand.
- Parsing extracts balanced JSON arrays/objects, supports single- and multi-series data, validates label/data length matches, rejects negative values, and guards against oversized or deeply nested input.
- `wrapLegacyChartShorthandLines` wraps matching lines in a `chart-shorthand` fence while skipping content already inside real fenced or indented code blocks (backtick, tilde, and nested fences).
- Added unit tests (`legacy-chart-shorthand.spec.ts`) covering detection, parsing edge cases, and fence-skipping behavior, plus render tests (`legacy-chart-shorthand.render.spec.tsx`) verifying single-point series render a circle marker instead of a zero-length polyline.
- Updated `generate-chart` action description and fallback hint to explicitly warn against typing its `title`/`labels`/`data`/`type` parameters as literal chat text, clarifying that the compatibility fallback is best-effort and does not re-query live data like the `/chart` embed.
- Added a changeset documenting the patch for `@agent-native/core`.

---

<a href="https://builder.io/app/projects/ae8cb7b8045d4c21b2a343e11036d454/cosmic-mystery-fcnf6yoh"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2F226fa21c49ce4f95a5aba53aa594fe7a">Edit in Builder</picture></a>&nbsp;&nbsp;<a href="https://ae8cb7b8045d4c21b2a343e11036d454-cosmic-mystery-fcnf6yoh.projects.builder.my/"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2Fe530b1333b5b4cedac9c41b8573c8268">Preview</picture></a>

---

To clone this PR locally use the [Github CLI](https://cli.github.com/) with command `gh pr checkout 2476`

You can tag me at @builderio for anything you want me to fix or change

### CODE REVIEW

**Comment by NKoech123** [comment]
@builderio-bot verify the above issues in comments exist, fix them and feel free to be skeptical. Fix the linting/formatting issue as well

File: packages/core/src/client/chat/legacy-chart-shorthand.tsx [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+  rawLine: string,
+): LegacyChartShorthand | null {
+  const trimmed = rawLine.trim();
+  if (!looksLikeLegacyChartShorthand(trimmed)) return null;
+
+  const typeMatch = trimmed.match(/\btype=(bar|line|area)\b/i);
+  const chartType =
+    (typeMatch?.[1].toLowerCase() as LegacyChartShorthand["type"]) || "bar";
+
+  const titleMatch = trimmed.match(/\btitle=(?:"([^"]*)"|'([^']*)')/);
```
[the remark this answers is not in the document]
Comment (NKoech123) [reply packages/core/src/client/chat/legacy-chart-shorthand.tsx]: @builderio-bot verify this issue exists and fix it

**Review by steve8708 — APPROVED** [review]
